File: public/docs/_examples/dependency-injection/ts/app/heroes/hero-list.component.2.ts
// #docregion
import { Component }   from '@angular/core';
import { Hero }        from './hero';
import { HeroService } from './hero.service';

@Component({
  selector: 'hero-list',
  template: `
  <div *ngFor="let hero of heroes">
    {{hero.id}} - {{hero.name}}
  </div>
  `,
})
export class HeroListComponent {
  heroes: Hero[];

  // #docregion ctor
  constructor(heroService: HeroService) {
    this.heroes = heroService.getHeroes();
  }
  // #enddocregion ctor
}

File: public/docs/_examples/dependency-injection/dart/lib/heroes/hero_list_component_2.dart
// #docplaster
// #docregion
import 'package:angular2/core.dart';

import 'hero.dart';
// #enddocregion
import 'hero_service_1.dart';
/*
// #docregion
import 'hero_service.dart';
// #enddocregion
*/
// #docregion

@Component(
    selector: 'hero-list',
    template: '''
      <div *ngFor="let hero of heroes">
        {{hero.id}} - {{hero.name}}
      </div>''')
class HeroListComponent {
  final List<Hero> heroes;

  // #docregion ctor
  HeroListComponent(HeroService heroService)
      : heroes = heroService.getHeroes();
  // #enddocregion ctor
}
